internal/k8s/controllers/frrk8s_config_controller_test.go

```diff
@@ -0,0 +1,169 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	frrv1beta1 "github.com/metallb/frr-k8s/api/v1beta1"
+	. "github.com/onsi/gomega"
+	v1beta1 "go.universe.tf/metallb/api/v1beta1"
+	v1beta2 "go.universe.tf/metallb/api/v1beta2"
+	frrk8s "go.universe.tf/metallb/internal/bgp/frrk8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+func TestFRRK8sReconcileReconciler_SetupWithManager(t *testing.T) {
+	g := NewGomegaWithT(t)
+	testEnv := &envtest.Environment{
+		// we should keep the frrk8s up to date. However, for the purpouse of testing the controller
+		// it is fine to keep this version and change one stable field (the router's asn).
+		CRDDirectoryPaths:     []string{filepath.Join("../../..", "config", "crd", "bases"), "./testdata"},
+		ErrorIfCRDPathMissing: true,
+		Scheme:                scheme,
+	}
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	defer func() {
+		err = testEnv.Stop()
+		g.Expect(err).ToNot(HaveOccurred())
+	}()
+	err = v1beta1.AddToScheme(k8sscheme.Scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = v1beta2.AddToScheme(k8sscheme.Scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	err = frrv1beta1.AddToScheme(k8sscheme.Scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+	m, err := manager.New(cfg, manager.Options{Metrics: metricsserver.Options{BindAddress: "0"}})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	r := &FRRK8sReconciler{
+		Client:    m.GetClient(),
+		Logger:    log.NewNopLogger(),
+		Scheme:    scheme,
+		NodeName:  "node",
+		Namespace: testNamespace,
+		Reload:    make(chan event.GenericEvent),
+	}
+
+	err = r.SetupWithManager(m)
+	g.Expect(err).ToNot(HaveOccurred())
+	ctx := context.Background()
+	go func() {
+		err = m.Start(ctx)
+		g.Expect(err).ToNot(HaveOccurred())
+	}()
+
+	ns := corev1.Namespace{}
+	ns.Name = testNamespace
+	err = m.GetClient().Create(ctx, &ns)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	frrConfig := frrv1beta1.FRRConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      frrk8s.ConfigName("node"),
+			Namespace: testNamespace,
+		},
+		Spec: frrv1beta1.FRRConfigurationSpec{
+			BGP: frrv1beta1.BGPConfig{
+				Routers: []frrv1beta1.Router{
+					{
+						ASN: 25,
+					},
+				},
+			},
+		},
+	}
+
+	err = m.GetClient().Create(ctx, &frrConfig)
```

Comment:
can you reload and eventually instead to see it creates properly on first time? also currently this might break the controller a bit as the desiredConfig is empty and it would attempt to reset everything (including the name/ns)?